packages/workflow/src/converter/data-converter.ts

```diff
@@ -88,16 +88,21 @@ export class CompositeDataConverter implements DataConverter {
   }
 }
 
-export function arrayFromPayloads(converter: DataConverter, content?: Payload[] | null): any[] {
+export async function arrayFromPayloads(converter: DataConverter, content?: Payload[] | null): Promise<unknown[]> {
```

Comment:
Is there a reason to make this function async? It doesn't use `await` and returns a promise, so all this `async` declaration does is create an unnecessary promise.

Reply:
The reason would be to have it included in the stack trace in case of error.

```js
async function b() {
  await new Promise(r => setTimeout(r, 10));
  throw new Error('heh');
}
function a() {
  return Promise.all([b()]);
}
a();
```

```
> Uncaught Error: heh
    at b (REPL48:3:9)
    at async Promise.all (index 0)
```

But for that to happen we need to await in `a` which we don't :/
```js
async function a() {
 return await Promise.all([b()]);
}
```

```
Uncaught Error: heh
    at b (REPL48:3:9)
    at async Promise.all (index 0)
    at async a (REPL62:2:10)
```

I'll add the `await` keyword, sounds good?